app/src/main/java/de/westnordost/streetcomplete/quests/barrier_opening/AddBarrierOpening.kt

```diff
@@ -0,0 +1,72 @@
+package de.westnordost.streetcomplete.quests.barrier_opening
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.geometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.WHEELCHAIR
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.quests.width.AddWidthForm
+import de.westnordost.streetcomplete.quests.width.WidthAnswer
+import de.westnordost.streetcomplete.screens.measure.ArSupportChecker
+
+class AddBarrierOpening(
+        private val checkArSupport: ArSupportChecker
+    ) : OsmElementQuestType<WidthAnswer> {
+
+    private val nodeFilter by lazy { """
+       nodes with
+         barrier ~ ${GATEWAYS.joinToString("|")}
```

Comment:
Relating to my comment in `AddWidthForm`:

When GATEWAYS is not used in any other file, which it shouldn't, IMO, then you should write it here like this

barrier ~ gate|entrance| ...etc

and delete the GATEWAYS field.

Reply:
Changes made